client/components/user/Oodlet.vue

```diff
@@ -46,57 +47,55 @@
 </template>
 
 <script>
-  import QuantifiedThingy from './QuantifiedThingy.vue';
   import moment from 'moment';
 
+  import QuantifiedThingy from './QuantifiedThingy.vue';
+
   export default {
     data() {
       return {
-        now : moment()
-      }
+        now: moment()
+      };
     },
     computed: {
       quantifiedThingies() {
         return this.$store.getters.quantifiedThingies;
       },
       dueDate() {
-        if(this.$store.getters.dueDate) {
-          return moment(this.$store.getters.dueDate);
-        }
+        return this.$store.getters.dueDate;
       },
       dueIn() {
-        return(
-              Math.trunc(this.dueDate.diff(this.now, 'days')) + 'd ' +
-              Math.trunc(this.dueDate.diff(this.now, 'hours')%24) + 'h ' +
-              Math.trunc(this.dueDate.diff(this.now, 'minutes')%60) + 'm ' +
-              Math.trunc(this.dueDate.diff(this.now, 'seconds')%60) + 's'
-        );
+        if (!this.dueDate) {
+          return '';
+        }
+
+        const dueDate = moment(this.dueDate);
+        const days = Math.trunc(dueDate.diff(this.now, 'days'));
+        const hours = Math.trunc(dueDate.diff(this.now, 'hours') % 24);
+        const minutes = Math.trunc(dueDate.diff(this.now, 'minutes') % 60);
+        const seconds = Math.trunc(dueDate.diff(this.now, 'seconds') % 60);
+        return `${days}d ${hours}h ${minutes}m ${seconds}s`;
       },
       dueInClass() {
```

Comment:
To me this is really confusing name. 'dueInClass' can refer to anything, it only returns colors as a computed property/string which is later gonna be used to display warning color according to time left, maybe just rename it to "timerColor" like we discussed together...